pkg/vault/contracts/Vault.sol

```diff
@@ -1353,45 +1360,94 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
+     * @notice Updates the reserves of the vault after an ERC4626 wrap (deposit/mint) operation.
+     * @dev If there are extra tokens in the vault balances, these will be added to the reserves (which, in practice,
+     * is equal to discarding such tokens). This approach avoids DoS attacks, when a frontrunner leaves vault balances
+     * and state of reserves out of sync before a transaction starts.
+     * @param underlyingToken Underlying of ERC4626 wrapped token
+     * @param wrappedToken ERC4626 wrapped token
+     * @param underlyingDeltaHint Amount of underlying tokens supposedly removed from the vault
+     * @param wrappedDeltaHint Amount of wrapped tokens supposedly added to the vault
      */
-    function _updateReservesAfterWrapping(
+    function _settleWrap(
         IERC20 underlyingToken,
-        IERC20 wrappedToken
-    ) internal returns (uint256 vaultUnderlyingDelta, uint256 vaultWrappedDelta) {
-        uint256 vaultUnderlyingBefore = _reservesOf[underlyingToken];
-        uint256 vaultUnderlyingAfter = underlyingToken.balanceOf(address(this));
-        _reservesOf[underlyingToken] = vaultUnderlyingAfter;
-
-        uint256 vaultWrappedBefore = _reservesOf[wrappedToken];
-        uint256 vaultWrappedAfter = wrappedToken.balanceOf(address(this));
-        _reservesOf[wrappedToken] = vaultWrappedAfter;
-
-        if (vaultUnderlyingBefore > vaultUnderlyingAfter) {
-            // Wrap
-            // Since deposit takes underlying tokens from the vault, the actual underlying tokens deposited is
-            // underlyingBefore - underlyingAfter
-            // checked against underflow: vaultUnderlyingBefore > vaultUnderlyingAfter in `if` clause.
-            unchecked {
-                vaultUnderlyingDelta = vaultUnderlyingBefore - vaultUnderlyingAfter;
-            }
-            // Since deposit puts wrapped tokens into the vault, the actual wrapped minted is
-            // wrappedAfter - wrappedBefore.
-            vaultWrappedDelta = vaultWrappedAfter - vaultWrappedBefore;
-        } else {
-            // Unwrap
-            // Since withdraw puts underlying tokens into the vault, the actual underlying token amount withdrawn is
-            // assetsAfter - assetsBefore.
-            // checked against underflow: vaultUnderlyingAfter > vaultUnderlyingBefore in `else` clause.
-            unchecked {
-                vaultUnderlyingDelta = vaultUnderlyingAfter - vaultUnderlyingBefore;
-            }
-            // Since withdraw takes wrapped tokens from the vault, the actual wrapped token amount burned is
-            // wrappedBefore - wrappedAfter.
-            vaultWrappedDelta = vaultWrappedBefore - vaultWrappedAfter;
+        IERC20 wrappedToken,
+        uint256 underlyingDeltaHint,
+        uint256 wrappedDeltaHint
+    ) internal {
+        // A wrap operation removes underlying tokens from the vault, so the expected vault's underlying balance after
```

Comment:
```suggestion
        // A wrap operation removes underlying tokens from the Vault, so the Vault's expected underlying balance after
```
Here and elsewhere. It's not strictly correct to do so, but we have been capitalizing "Vault" pretty much everywhere in the code. Should be done here too for consistency.